src/cls/lock/cls_lock.cc

```diff
@@ -37,6 +38,10 @@ CLS_NAME(lock)
 
 #define LOCK_PREFIX    "lock."
 
+/// name of the lock used on objects to ensure layout stability during IO
+/// reference from src/libradosstriper/RadosStriperImpl.cc
+#define RADOS_LOCK_NAME "striper.lock"
```

Comment:
would be better to import this from a common header, e.g. put this in cls_lock_types.h and include that in RadosStriperImpl.cc. It could be a const std::string instead of a #define as well.

It's a bit odd to embed knowledge of libradosstriper in the generic cls_lock implementation, but probably better than duplicating it. @dillaman do you have any suggestions?

Reply:
You could encode the limit as an optional in `cls_lock_lock_op`

Reply:
I found that `src/libradosstriper/RadosStriperImpl.cc` includes `cls/lock/cls_lock_client.h`, which includes `cls/lock/cls_lock_types.h`, so maybe I just add `RADOS_LOCK_NAME` to `cls/lock/cls_lock_types.h` will be ok.

Reply:
I like @dillaman's idea of adding the max locks as an optional argument - then there's no need for cls_lock to understand anything specific to libradosstriper